### PR TITLE
[CWS] add selftest for ebpfless

### PR DESCRIPTION
**Pull request by safchain**
### What does this PR do?

The self test event for the ebpfless mode is the first exec event.

### CODE REVIEW

**Review by spikat — APPROVED** [review]

**Review by mftoure — APPROVED** [review]

File: pkg/security/probe/selftests/tester_linux.go [hunk 1]
```diff
@@ -26,169 +25,212 @@ import (
...
+		t.success, t.fails, t.lastTimestamp = success, fails, time.Now()
+		t.Unlock()
 
-	// save the results for get status command
-	t.success = success
-	t.fails = fails
+		cb(success, fails, events)
 
-	return success, fails, testEvents, nil
+		t.endSelfTests()
```
Comment (YoannGh) [inline pkg/security/probe/selftests/tester_linux.go:191]: Shouldn't we break out of the loop here? So that we avoid reporting the result multiple times?

Comment (safchain) [reply pkg/security/probe/selftests/tester_linux.go:191]: we don't want to break as we want to handle self tests through gRPC as well

File: pkg/security/module/cws.go [hunk 2]
```diff
@@ -169,28 +181,24 @@ func (c *CWSConsumer) PostProbeStart() error {
 }
 
 // RunSelfTest runs the self tests
-func (c *CWSConsumer) RunSelfTest(sendLoadedReport bool) (bool, error) {
+func (c *CWSConsumer) RunSelfTest(gRPC bool) (bool, error) {
+	if c.config.EBPFLessEnabled {
```
Comment (YoannGh) [inline pkg/security/module/cws.go:185]:
```suggestion
	if c.config.EBPFLessEnabled && gRPC {
```

**Review by YoannGh — APPROVED** [review]

**Review by YoannGh — APPROVED** [review]

**Review by spikat — APPROVED** [review]

**Pull request merged**